File: 55.跳跃游戏.go
/*
 * @lc app=leetcode.cn id=55 lang=golang
 *
 * [55] 跳跃游戏
 */

// @lc code=start
func canJump(nums []int) bool {
	length := len(nums)
	dp := make([]bool,length)
	dp[length-1] = true
	for i := length-2 ; i >= 0 ; i-- {
		for j := nums[i] ; j > 0 ; j-- {
			if i + j >= length - 1{
				dp[i] = true
				break
			}else{
				if dp[i+j] == true {
					dp[i] = true
					break
				}
			}
		}
	}
	return dp[0]
}
// @lc code=end

File: 55.跳跃游戏.ts
/*
 * @lc app=leetcode.cn id=55 lang=typescript
 *
 * [55] 跳跃游戏
 */

// @lc code=start
function canJump(nums: number[]): boolean {
    if (nums.length === 1) {
        return true;
    }
    let dp: boolean[] = new Array<boolean>(nums.length).fill(false);
    dp[0] = true;
    for (let i = 0; i < nums.length - 1; i++) {
        if (!dp[i]) {
            continue;
        }
        for (let j = 1; j < nums.length && j <= i + nums[i]; j++) {
            dp[j] = true;
            if (j === nums.length - 1) {
                return true;
            }
        }
    }
    return false;
};
// @lc code=end

